dotnet/src/webdriver/CommandInfoRepository.cs

```diff
@@ -106,6 +109,12 @@ public T GetCommandInfo<T>(string commandName) where T : CommandInfo
         /// This method will not overwrite existing commands for a specific name, and will return <see langword="false"/>
         /// in that case.
         /// </remarks>
+        /// <exception cref="ArgumentNullException">
+        /// <para>If <paramref name="commandName"/> is <see langword="null"/> or <see cref="string.Empty"/>.</para>
+        /// <para>-or-</para>
+        /// <para>If <paramref name="commandInfo"/> is <see langword="null"/>.</para>
+        /// </exception>
+        /// <exception cref="ArgumentException">If <typeparamref name="T"/> is not a valid command type for this repository.</exception>
```

Comment:
`ArgumentOutOfRangeException` is better? Just to double-check, I trust you.

Reply:
Personally I would make this an `InvalidOperationException`. Even better, I would make this class generic on `TCommandInfo` and pass in that value. (this would also remove the need for the `Type RepositoryCommandInfoType` property).

However, both of those are breaking changes. This type is `public` and I can only assume someone somewhere may be using this. Since the exception is based on invalid input, `ArgumentException` isn't necessarily wrong.